From 43973650556fdab7dc44746b2c87a204c3a94753 Mon Sep 17 00:00:00 2001
Message-Id: <43973650556fdab7dc44746b2c87a204c3a94753.1432858002.git.chang-joon.lee@intel.com>
In-Reply-To: <944dfa31d45b9ee101932dfa029ec3452a83cb30.1432858002.git.chang-joon.lee@intel.com>
References: <944dfa31d45b9ee101932dfa029ec3452a83cb30.1432858002.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Tue, 24 Mar 2015 19:28:48 +0530
Subject: [PATCH 3/7] SQUASHME! [VPG]: drm/i915/vlv: Remove unnecessary mutex
 around rc6 enable.

The rps mutex lock around valleyview_enable_rc6 not needed as
we do not do any sideband access during rc6 feature enable.

SQUASHME! - This patch should be squashed into the following existing patch:
commit fae41e76d7ccb060283f72fcbdf7743c0b4775b3
change-Id: I2872bb45bd76cf0d8cade3737a1b44e6bf9dab86
Author: Sagar Kamble <sagar.a.kamble@intel.com>
Date:   Thu Nov 13 14:12:43 2014 +0530

    REVERTME [VPG]: drm/i915/vlv: Enabling RC6 immediately during
    init/resume

Issue: GMINL-7571
Change-Id: I4fed87426d8fe2314e840c8352d579cba0ab0128
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index 88838b5..c5d81fa 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -3799,8 +3799,6 @@ void vlv_set_rc6_mode(struct drm_device *dev, bool disable)
 		return;
 	}
 
-	WARN_ON(!mutex_is_locked(&dev_priv->rps.hw_lock));
-
 	dev_priv->rps.rc6_disable = disable;
 
 	if (disable)
@@ -4965,8 +4963,6 @@ static void valleyview_enable_rc6(struct drm_device *dev)
 	u32 gtfifodbg, rc6_mode = 0, pcbr;
 	int i;
 
-	WARN_ON(!mutex_is_locked(&dev_priv->rps.hw_lock));
-
 	valleyview_check_pctx(dev_priv);
 
 	if ((gtfifodbg = I915_READ(GTFIFODBG))) {
@@ -5903,9 +5899,7 @@ void intel_enable_gt_powersave(struct drm_device *dev)
 		 * enabling.
 		 */
 		if (IS_VALLEYVIEW(dev) && !IS_CHERRYVIEW(dev)) {
-			mutex_lock(&dev_priv->rps.hw_lock);
 			valleyview_enable_rc6(dev);
-			mutex_unlock(&dev_priv->rps.hw_lock);
 		}
 
 		/*
-- 
1.9.1

